### DISCUSSION TITLE
Wie habt ihr Eure Anlage konfiguriert ?

### DISCUSSION THREAD

**Discussion by aruttkamp** [aruttkamp/ioBroker.rct#43] [General]
Habt Ihr eure Anlage selbst konfiguriert ? ( Stichwort SOC ) 
Was überwacht Ihr ? 
Welche Schlüsse zieht ihr daraus ? 

Würde mich freuen wenn wir hier ein paar Dinge sammeln, die dann anderen helfen können ! 

@NCIceWolf @wiese-computer-brb @BenS89 @camast

**Comment by aruttkamp** [comment]
Ich habe an der Grundkonfiguration bisher nichts geändert. Akku wird bis 7% entladen und bis 97% geladen. 
Überwachen tue ich den aktuellen Hausverbrauch, die aktuelle Produktion, sowie den Stand des Akkus. 
Bisher schalte ich auch keine Geräte in Abhängigkeit der Stati.

Habe für mich erstmal Daten gesammelt und kann sagen.. bis gestern war ich recht autark. Immer vorrausgesetzt dass nicht nach Sonnenuntergang gewaschen, gekocht o.ä. wird ;-)

**Comment by BenS89** [comment]
@aruttkamp Vielen Dank, dass du den Adapter weiterentwickelst.
Ich nutze deinen Adapter, um diverse Werte über den ioBroker in eine InfluxDB zu schreiben und dann via Grafana anzuzeigen. 

Dazu gehören:
- Aktuelle Erzeugung jetzt/Tag/Monat/Jahr
- Aktueller Verbrauch jetzt/Tag/Monat/Jahr
- Aktuelle Einspeisung jetzt/Tag/Monat/Jahr
- Autarkiegrad Tag/Monat/Jahr
- Stromverbrauch Haus auf Zeitleiste
- Batterieladekurve (inkl. Telegram Push, falls der SOC <7% oder > 97 % gehen sollte) --> So kann man SOC-Sprünge erkennen, wenn welche vorkommen --> meine Batterie muss deswegen getauscht werden
- Zyclenzahl der Batterie
- Stromkosten und Netzeinspeise-Vergütung auf Basis der aktuellen kw-Preise

Zudem habe ich Blockly´s Laufen, um bestimmte Dinge mit dem Stromüberschuss zu machen, z.B.
- Spülmaschine einschalten
- Warmwasser der Wärmepumpe erzeugen

2 Fragen an euch: 

1. Welche Wechselricher Firmware habt ihr installiert? Ich habe aktuell einen RCT Release Candidate drauf (noch nicht veröffentlicht) und derzeit das Problem, dass die Batterie beim erreichen des SOC von 97% nicht idle ist, sondern einen Strom-Abfluss von 23-26 Watt hat. Das erscheint mir nicht korrekt zu sein

2. Gibt es eine Möglichkeit, irgendwo die Gesamtkapazität des Akkus auszulesen? Also alle kWh der einzelnen Module aufsummiert? Bisher habe ich keinen Parameter hierzu gefunden

**Comment by aruttkamp** [comment]
Moin ! 
1. Welche Wechselricher Firmware habt ihr installiert? Ich habe aktuell einen RCT Release Candidate drauf (noch nicht veröffentlicht) und derzeit das Problem, dass die Batterie beim erreichen des SOC von 97% nicht idle ist, sondern einen Strom-Abfluss von 23-26 Watt hat. Das erscheint mir nicht korrekt zu sein

**Control Software Version 2.3.5661**

2. Gibt es eine Möglichkeit, irgendwo die Gesamtkapazität des Akkus auszulesen? Also alle kWh der einzelnen Module aufsummiert? Bisher habe ich keinen Parameter hierzu gefunden

Habe ich bisher nicht gefunden. Weder in Summe - noch einzeln.

**Reply by NCIceWolf** [reply]
Moin!

Welche Wechselricher Firmware habt ihr installiert?
-->
Control Software Version: 2.3.5661
BMS-Version: 5599

Gibt es eine Möglichkeit, irgendwo die Gesamtkapazität des Akkus auszulesen? Also alle kWh der einzelnen Module aufsummiert?
-->
Hab ich ebenfalls nicht gefunden. Aber den errechneten Wirkungsgrad gibt es wohl 😄

**Comment by aruttkamp** [comment]
Noch ne Frage @BenS89 
>>Spülmaschine einschalten

schaltest nur ne Steckdose auf "an" oder kann Deine Waschmaschine dann auch von alleine loslaufen ?

**Reply by BenS89** [reply]
Moin! Die Steuerung der Spülmaschine läuft über die HomeConnect API.

`battery.ah_capacity` könnte interessant sein. Bei mir kommt da aber Crap raus - ich vermute dass auch hier der Datentyp noch nicht passt.

**Comment by aruttkamp** [comment]
*patsch* Spülmaschine ... alles klar. das kann ich auch machen. ich hatte irgendwie "Waschmaschine" gelesen.

battery.ah_capacity liefert bei mir den Wert 25 Ah

**Reply by BenS89** [reply]
Bei mir steht da 22,2 drin. 

Folgendes hatte ich mal gelesen --> Das liest sich interessant. Könnte das in unserem Fall noch ein falscher Datentyp sein?

> Es gibt "battery.ah_capacity". Nach meinem Eindruck wird dort die aktuelle Bruttokapazität ausgegeben.
Meine Batterie ist brutto ab Werk mit 7,5 kWh angegeben. Nach 3 Jahren Betrieb gibt dieser Parameter bei mir 7,3 kWh zurück.
Die Abfrage mache ich aktuell über rctclient, verpackt in einen kleinen wrapper.

**Reply by aruttkamp** [reply]
Bin auch kein Elektriker, aber ich hab da mal gegoogelt ;-) 

der Wert heisst battery.**ah**_capacity  und wird auch mit der Einheit Ah bezeichnet. 
der kann eigentlich keinen kWh Wert liefern. 

[Ah zu kWh ist eine Umrechnung von Amperestunden zu Kilowattstunden, die die Kapazität und die Energie eines Akkus beschreiben. Um Ah zu kWh umzurechnen, muss man die Spannung des Akkus in Volt oder Kilovolt kennen. Die Formel lautet kWh = Ah x V / 1000 oder kWh = kV x Ah]

Ergäbe bei 25 Ah und 320,1V 
25*320,1/1000 = 8,0025 kw

Benutzt habe ich die Werte battery.ah_capacity und battery.voltage.
Der Wert erscheint mir für eine neue Batterie akzeptabel ;-)

**Comment by NCIceWolf** [comment]
Moin!

Mache ebenfalls viel über Grafana, also hauptsächlich Anzeigen. Mir gefällt einfach die Historie in der RCT App nicht, das hab ich mir lieber schöner in Grafana gebaut.
Die einzigen Steuerungen die ich mache betreffen die Außenbeleuchtung (bei genug Akku und Vorhersage für den nächsten Tag) und die Wallbox. Und da nutze ich den Akkustand um Bewölkung auszugleichen, also trotzdem weiterzuladen obwohl gerade weniger Energie verfügbar ist als die 1,4kW die ins Auto müssen.

Ein Punkt zu dem ich noch etwas Gehirnschmalz investieren will ist das Thema Batteriekonditionierung. Hier versuche ich aktuell meinen Verbrauch künstlich zu erhöhen nachdem der Akku auf 100% ist. Umso früher der Akku dann leer ist, umso eher kriegt man noch was in den Akku rein bevor die Sonne untergeht. Am sinnvollsten schaffe ich das bisher mit dem Auto. So versuche ich externen Bezug in der Nacht zu verhindern.

**Reply by aruttkamp** [reply]
Schau meine Antwort an BenS89 - Sollte die Geamtkapazität sein. Wenn ich alles richtig verstanden habe.

**Reply by OnkelFD** [reply]
Schau mal in die Diskussion https://github.com/aruttkamp/ioBroker.rct/discussions/69 den Datenpunkt kannst du verwenden wenn du wissen willst was das Batterie Management vor hat.

**Comment by aruttkamp** [comment]
@BenS89 
Zudem habe ich Blockly´s Laufen, um bestimmte Dinge mit dem Stromüberschuss zu machen, z.B.
Spülmaschine einschalten

Magst das Blockly zur Verfügung stellen ? Wir könnten hier solche Skripte sammeln.

**Reply by BenS89** [reply]
Klar, kein Problem. Mache ich, sobald ich etwas Luft habe

**Comment by montee6** [comment]
Ich habe eine PV mit 10,53 kWp, 8KW RCT Wechselricher und 7,6kWh Speicher. 
Dazu eine Wärmepumpe (mit zusätzlicher Brennwerttherme wenn es draußen zu kalt ist). 
Die Wärmepumpe hat einen extra Eingang für einen Strompreis2 den lasse ich allerdings nicht direkt vom RCT Wechselrichter schalten, sondern habe mir eine Elektronik gebaut (mit ESP32), die via MQTT und einem IoBroker Script diesen Eingang steuert und zusätzlich die Temperaturen im Speicherkessel und die Außentemperatur misst.
Normalerweise, also ohne Script würde die Wärmepumpe mit Strompreis2 den Kessel auf 50°C aufheizen.
Bei niedrigen Temperaturen ist das ineffizient und zieht zuviel Strom. 
Das Script wertet deshalb die Außentemperatur aus und legt eine sinnvolle Maximale Kesseltemperatur fest bei der die Wärmepumpe noch effizient heizt, und bei deren Erreichen der Strompreis2 wieder abgeschaltet wird.

Morgens (bzw. immer) wertet das Script den **SOC** der Batterie aus, schaltet bei Erreichen eines bestimmten Ladezustands (und mindest **PV power**) den Strompreis2 ein. Im Vis habe ich mir dafür ein minSOC Eingabefeld geschaffen, damit ich auch mal abends mehr die Wärmepumpe bemühen kann, wenn ich weiß, dass der nächste Tag gleich mit Sonne beginnt. (Das könnte ich auch noch Scripten wenn Zeit dafür ist.)

**Comment by BenS89** [comment]
> @BenS89 Zudem habe ich Blockly´s Laufen, um bestimmte Dinge mit dem Stromüberschuss zu machen, z.B. Spülmaschine einschalten
> 
> Magst das Blockly zur Verfügung stellen ? Wir könnten hier solche Skripte sammeln.

**Comment by aruttkamp** [comment]
Mag wirklich keiner seine Blockly´s zeigen ?  
Schade eigentlich !

**Reply by BenS89** [reply]
Na klar! Anbei mal meine Logik zur Warmwasser-Steuerung in Abhängigkeit des PV-Überschusses. Ich bin auf eure Skripte gespannt!


<img width="634" alt="Warmwasser" src="https://github.com/aruttkamp/ioBroker.rct/assets/5046137/2f7bc5a9-0237-486d-b3da-774760cbbdd6">

**Reply by NCIceWolf** [reply]
Musste erst einmal den Wildwuchs bekämpfen und ein paar Kommentare einblenden 😄 

Steuere darüber meine Klimaanlagen für das Kinderzimmer, Büro und Wohnzimmer, z.B.:

![Blockly_Klima-Matteo](https://github.com/aruttkamp/ioBroker.rct/assets/127289023/f133c76f-02c0-4210-9748-5678106df340)
![Blockly_Klima-Büro](https://github.com/aruttkamp/ioBroker.rct/assets/127289023/bd936a94-bf6a-45d2-986e-5a004e4c4bed)

**Reply by NCIceWolf** [reply]
Dazu habe ich ein recht umfangreiches Skript für meine Wallbox gebaut (Easee), inklusive 1-3-Phasenumschaltung.
Das ist nur leider viel zu groß um es (lesbar) hier einzufügen.

Sollte da jemand Interesse haben, kann ich das Skript aber gerne exportieren und hier einfügen.
